p4-16/spec/P4-16-spec.mdk

```diff
@@ -4810,6 +4839,30 @@ the time budget allocated for parsing, the parser should transition to
 `reject` and set
 the standard error `error.ParserTimeout`.
 
+##  Parser Value Sets { #sec-value-set }
+
+In some cases, the values that determine the transition from one parser state
+to another need to be determined at run time. MPLS is one example where the
+value of the MPLS label field is used to determine what headers follow the MPLS
+tag and this mapping may change dynamically at run time. To support this
+functionality, P4 supports the notion of a Parser Value Set. This is a named
+set of values with a run time API to add and remove values from the set.
+
+Value sets are declared locally within a parser. They should be declared before
+being referenced in parser `keysetExpression`.
+
+Parser Value Sets support a `size` argument to provide hints to the compiler to
```

Comment:
Maybe you can either cross-reference the previous section or give another example here.